worker/src/tests/synchronizer_tests.rs

```diff
@@ -14,6 +14,7 @@ use tokio::time::timeout;
 async fn synchronize() {
```

Comment:
Currently 1 and 2 are pretty much the same case, since errors to the `WorkerBatchRequest` are treated the same as timeouts/non-responses--the existing timer-based retry mechanism (which remains unchanged) activates and sends a `lucky_broadcast` request to a few other random nodes for the same batches.

Given that the retry mechanism will soon change in the next PR when I refactor the `Synchronize` call coming from the primary, would it be fine by you if I hold of adding this test until then? I think otherwise some of it will have to be immediately rewritten.

Reply:
(Note that one benefit of the next step in refactoring is that we can immediately retry requests for batches that the initial target worker does not have, instead of having to wait for a timeout)